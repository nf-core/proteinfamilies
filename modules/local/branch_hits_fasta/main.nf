process BRANCH_HITS_FASTA {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/eb/eb3700531c7ec639f59f084ab64c05e881d654dcf829db163539f2f0b095e09d/data' :
        'community.wave.seqera.io/library/biopython:1.84--3318633dad0031e7' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(domtbl)
    val(length_threshold)

    output:
    tuple val(meta), path("hits/*")    , emit: hits
    tuple val(meta), path("*.fasta.gz"), emit: non_hit_fasta
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    branch_hits_fasta.py \\
        --fasta ${fasta} \\
        --domtbl ${domtbl} \\
        --length_threshold ${length_threshold} \\
        --hits hits \\
        --non_hits ${prefix}.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """
}
