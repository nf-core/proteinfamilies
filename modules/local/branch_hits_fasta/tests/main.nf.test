nextflow_process {

    name "Test Process BRANCH_HITS_FASTA"
    script "../main.nf"
    process "BRANCH_HITS_FASTA"

    test("proteinfamilies - fasta") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'fasta' ],
                    file(params.pipelines_testdata_base_path + 'test_data/mgnifams_input_small.faa', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'domtbl' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test.domtbl.gz', checkIfExists: true)
                ])
                input[2] = 0.9
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.hits,
                    process.out.non_hit_fasta,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'fasta' ],
                    file(params.pipelines_testdata_base_path + 'test_data/mgnifams_input_small.faa', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'domtbl' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test.domtbl.gz', checkIfExists: true)
                ])
                input[2] = 0.9
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
