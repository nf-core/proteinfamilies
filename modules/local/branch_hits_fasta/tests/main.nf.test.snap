{
    "proteinfamilies - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "fasta"
                        },
                        "test.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "fasta"
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,c8b64849d0e062bd2ec6db470afe03d0"
                ],
                "hits": [
                    [
                        {
                            "id": "fasta"
                        },
                        "test.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "non_hit_fasta": [
                    [
                        {
                            "id": "fasta"
                        },
                        "test.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c8b64849d0e062bd2ec6db470afe03d0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-06-13T14:13:49.253831461"
    },
    "proteinfamilies - fasta": {
        "content": [
            [
                [
                    {
                        "id": "fasta"
                    },
                    [
                        "mgnifams_test_1.fasta:md5,df53aa6a0450be4f1c73c51578ba5323",
                        "mgnifams_test_3.fasta:md5,25bc45d9938415d996e1ff0fef235f78",
                        "mgnifams_test_4.fasta:md5,4f8efff250f1242767d6eaab614b880f",
                        "mgnifams_test_5.fasta:md5,666cf3c54399d0088fd5ad03eec6e575",
                        "mgnifams_test_6.fasta:md5,c3ac1f49721a3d5faa69ec916aab29d2",
                        "mgnifams_test_7.fasta:md5,161d18858155e5185ff3b1900d58cf48"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "fasta"
                    },
                    "fasta.fasta.gz:md5,7ab7e32d9e439bba8ee151cd2d278994"
                ]
            ],
            [
                {
                    "BRANCH_HITS_FASTA": {
                        "python": "3.13.1",
                        "biopython": 1.85
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-06-13T14:13:37.087871575"
    }
}